File: exercises/day13-Keyboard-First-Refactoring/scala/src/main/scala/sleightrip/elf/Elf.scala
package sleightrip.elf

import sleightrip.trip.SleightTrip
import scala.collection.mutable.ListBuffer

/**
 * Represents an elf in Santa's workshop.
 *
 * Each elf:
 *  - has a list of workshop friends (other elves)
 *  - has a list of sleigh trips they went on to deliver presents
 */
class Elf {
  private val sleighTrips = ListBuffer[SleightTrip]()
  private val friends = ListBuffer[Elf]()

  def getFriends: List[Elf] = friends.toList

  def addFriend(elf: Elf): Unit = {
    friends += elf
  }

  def addTrip(sleighTrip: SleightTrip): Unit = {
    sleighTrips += sleighTrip
  }

  def trips(): List[SleightTrip] = sleighTrips.toList
}

File: exercises/day13-Keyboard-First-Refactoring/kotlin/src/main/kotlin/sleightrip/elf/Elf.kt
package sleightrip.elf

import sleightrip.trip.SleightTrip

/**
 * Represents an elf in Santa's workshop.
 *
 * Each elf:
 *  - has a list of workshop friends (other elves)
 *  - has a list of sleigh trips they went on to deliver presents
 */
class Elf {
    private val sleighTrips = mutableListOf<SleightTrip>()
    private val friends = mutableListOf<Elf>()

    fun getFriends(): List<Elf> = friends

    fun addFriend(elf: Elf) {
        friends.add(elf)
    }

    fun addTrip(sleighTrip: SleightTrip) {
        sleighTrips.add(sleighTrip)
    }

    fun trips(): List<SleightTrip> = sleighTrips
}
